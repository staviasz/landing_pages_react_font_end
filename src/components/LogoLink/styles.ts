import styled, { css } from 'styled-components';

export const LinkContent = styled.a`
  ${() => css`
    display: flex;
    align-items: center;
    text-decoration: none;
    color: inherit;
    > img {
      max-height: 3rem;
    }
  `}
`;
