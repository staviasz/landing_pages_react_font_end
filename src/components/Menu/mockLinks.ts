export const links = [
  {
    children: 'link1',
    href: '#target1',
    target: '_self',
  },
  {
    children: 'link2',
    href: '#target2',
    target: '_self',
  },
  {
    children: 'link3',
    href: '#target3',
    target: '_self',
  },
  {
    children: 'link4',
    href: '#target4',
    target: '_self',
  },
  {
    children: 'link5',
    href: '#target5',
    target: '_self',
  },
  {
    children: 'link6',
    href: '#target6',
    target: '_self',
  },
  {
    children: 'link7',
    href: '#target7',
    target: '_self',
  },
  {
    children: 'link8',
    href: '#target8',
    target: '_self',
  },
  {
    children: 'link9',
    href: '#target9',
    target: '_self',
  },
  {
    children: 'link10',
    href: '#target10',
    target: '_self',
  },
  {
    children: 'link11',
    href: '#target11',
    target: '_self',
  },
  {
    children: 'link12',
    href: '#target12',
    target: '_self',
  },
  {
    children: 'link13',
    href: '#target13',
    target: '_self',
  },
  {
    children: 'link14',
    href: '#target14',
    target: '_self',
  },
  {
    children: 'link15',
    href: '#target15',
    target: '_self',
  },
  {
    children: 'link16',
    href: '#target16',
    target: '_self',
  },
];
