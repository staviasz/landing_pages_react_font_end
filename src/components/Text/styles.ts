import styled, { css } from 'styled-components';

export const Text = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xsmall};
    text-align: left;
  `}
`;
