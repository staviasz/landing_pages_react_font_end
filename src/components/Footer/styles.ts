import styled, { css } from 'styled-components';
import { Container as SectionContainer } from '../SectionContainer/styles';
import { Text } from '../Text/styles';

export const FooterContainer = styled.footer`
  ${({ theme }) => css`
    border-top: 0.1rem solid ${theme.colors.mediumGray};

    > ${SectionContainer} {
      padding: ${theme.spacings.xsmall};
      @media ${theme.media.lteMedium} {
        max-width: 80%;
        margin-left: 18px;
      }
    }

    ${Text} {
      font-size: ${theme.font.sizes.small};
      text-align: center;
      margin: 0;
    }
    a {
      color: inherit;
      text-decoration: none;
      font-size: ${theme.font.sizes.small};
    }
  `}
`;
