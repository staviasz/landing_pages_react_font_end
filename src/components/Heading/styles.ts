import styled, { css } from 'styled-components';
import { Theme } from '../../styles/theme';

interface TitleProps {
  colordark: string;
  size: string;
  uppercase: string;
}

interface TitleSize {
  [key: string]: (theme: Theme) => ReturnType<typeof css>;
}

const mediaFont = (theme: Theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.xlarge};
  }
`;

const titleSize: TitleSize = {
  small: (theme) => css`
    font-size: ${theme.font.sizes.medium};
  `,
  medium: (theme) => css`
    font-size: ${theme.font.sizes.large};
  `,
  big: (theme) => css`
    font-size: ${theme.font.sizes.xlarge};
  `,
  xhuge: (theme) => css`
    ${mediaFont(theme)}
    font-size: ${theme.font.sizes.xhuge};
  `,
};

export const Title = styled.h1<TitleProps>`
  ${({ theme, colordark, size, uppercase }) => css`
    color: ${colordark === 'dark'
      ? theme.colors.primaryColor
      : theme.colors.white};
    ${titleSize[size](theme)}
    text-transform: ${uppercase};
    font-family: ${theme.font.family.secondary};
    font-style: italic;
  `}
`;
