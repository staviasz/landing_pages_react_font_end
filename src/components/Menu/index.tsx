import { Close as CloseIcon } from '@styled-icons/material-outlined/Close';
import { Menu as MenuIcon } from '@styled-icons/material-outlined/Menu';
import { useState } from 'react';
import { LogoLink, LogoLinkProps } from '../LogoLink';
import { NavLink, NavLinkProps } from '../NavLinks';
import { SectionContainer } from '../SectionContainer';
import { Button, Container, MenuContainer } from './styles';

interface MenuProps extends NavLinkProps {
  children?: string | React.ReactNode;
  logoData: LogoLinkProps;
}

export const Menu = ({ links = [], logoData }: MenuProps) => {
  const [visible, setVisible] = useState(false);

  return (
    <>
      <Button
        visible={visible}
        onClick={() => setVisible(true)}
        aria-label="Open/Close menu"
      >
        {visible ? (
          <CloseIcon aria-label="Close menu" />
        ) : (
          <MenuIcon aria-label="Open menu" />
        )}
      </Button>
      <Container visible={visible} onClick={() => setVisible(false)}>
        <SectionContainer>
          <MenuContainer>
            <LogoLink {...logoData} />
            <NavLink links={links} />
          </MenuContainer>
        </SectionContainer>
      </Container>
    </>
  );
};
