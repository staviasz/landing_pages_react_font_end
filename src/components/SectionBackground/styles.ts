import styled, { css } from 'styled-components';
import { Theme } from '../../styles/theme';

interface ContainerProps {
  background: string;
}

const containerBackgroundActivate = (theme: Theme) => css`
  background: ${theme.colors.primaryColor};
  color: ${theme.colors.white};
`;

export const Container = styled.div<ContainerProps>`
  ${({ theme, background }) => css`
    background: ${theme.colors.white};
    color: ${theme.colors.primaryColor};
    ${background === 'true' && containerBackgroundActivate(theme)};
    min-height: 100vh;
    display: flex;
    align-items: center;
  `}
`;
