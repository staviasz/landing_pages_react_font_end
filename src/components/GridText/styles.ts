import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';
import { Text } from '../Text/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${Text} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${() => css`
    ${Title} {
      position: relative;
      left: 5rem;
    }
    ${Title}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: 7rem;
      top: -3rem;
      left: -5rem;
      transform: rotate(5deg);
    }
  `}
`;
