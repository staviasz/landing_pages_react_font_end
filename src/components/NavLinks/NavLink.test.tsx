import { screen } from '@testing-library/react';
import { NavLink } from '.';
import { renderTheme } from '../../styles/render-theme';
import { links } from '../Menu/mockLinks';

describe('<NavLink/>', () => {
  it('should render', () => {
    renderTheme(<NavLink links={links}></NavLink>);
    expect(screen.getAllByRole('link')).toHaveLength(links.length);
  });

  it('should not render links', () => {
    renderTheme(<NavLink />);
    expect(screen.queryAllByText(/link/i)).toHaveLength(0);
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<NavLink links={links}></NavLink>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
