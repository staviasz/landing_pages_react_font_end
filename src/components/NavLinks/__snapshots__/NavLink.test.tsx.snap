// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLink/> should match snapshot 1`] = ` `;
